README.md

```diff
@@ -606,6 +606,81 @@ Output is written to stdout.
       future obfuscation with --load_map.); default: "";
 ```
 
+## Indexing Verible C++ Code using Kythe
+
+The steps mentioned here can be generalized for indexing Bazel-based projects.
+
+### Initializing Kythe
+Download the latest Kythe release from https://github.com/kythe/kythe/releases and then unpack it for a snapshot of Kythe’s toolset.
+
+```bash
+tar xzf kythe-v*.tar.gz
+rm -rf /opt/kythe
+mv kythe-v*/ /opt/kythe
+```
+More information can be found [here](https://github.com/kythe/kythe#getting-started).
+
+Clone Kythe from [here](https://github.com/kythe/kythe). Then from within the Kythe clone, build the web frontend and copy its files into /opt/kythe/
+
+```bash
+bazel build //kythe/web/ui
+mkdir -p /opt/kythe/web/ui
+cp -r bazel-bin/kythe/web/ui/resources/public/* /opt/kythe/web/ui
+cp -r kythe/web/ui/resources/public/* /opt/kythe/web/ui
+chmod -R 755 /opt/kythe/web/ui
+```
+More information can be found [here](https://github.com/google/haskell-indexer#building-from-source).
+
+### Extracting Compilations for Verible
+
+#### VNames.json File
+`vnames.json` file is used for renaming certain filepaths during extraction, but renaming isn’t needed here but you can add the renaming you find suitable.
+
+#### Run the extractor
+```bash
+# run on all targets
+bazel test --experimental_action_listener=:extract_cxx  //...
+
+# run on specific target (e.g. some cc_binary)
+bazel test --experimental_action_listener=:extract_cxx //verilog/analysis:default_rules
+```
+
+Extracted kzip files will be in bazel-out/local-fastbuild/extra_actions/extractor folder. One kzip file per target.
+```bash
+find -L bazel-out -name '*cxx.kzip'
+```
+More information can be found [here](https://github.com/kythe/kythe/tree/master/kythe/cxx/extractor).
+
+### Indexing Extracted Kzip Files & Generating GraphStore
+For extracting the indexing facts from extracted `kzip` files, run the following command to apply the indexing for each `kzip` file and generate the result into `kythe graphstore`.
+```bash
+for i in $(find -L bazel-out -name '*cxx.kzip'); do
+    # Indexing C++ compilations
+    /opt/kythe/indexers/cxx_indexer --ignore_unimplemented "$i" > entries
+
+    # Write entry stream into a GraphStore
+    /opt/kythe/tools/write_entries --graphstore leveldb:.kythe_graphstore < entries
+done
+```
+
+### Generating Serving Tables
+```bash
+# Generate corresponding serving tables
+/opt/kythe/tools/write_tables --graphstore .kythe_graphstore --out .kythe_serving
+```
+
+### Visualizing Cross-References
+At this point we got Kythe’s serving tables and this can be used for visualising cross-referencing and code navigation using `Kythe’s ui`
```

Comment:
Suggest phrasing:

"Run Kythe's UI to visualize cross-reference and code navigation."